backend/tests/unit/controllers/posts-controller.test.js

```diff
@@ -0,0 +1,312 @@
+import {
+  createPostHandler,
+  getAllPostsHandler,
+  getFeaturedPostsHandler,
+  getPostByCategoryHandler,
+  getLatestPostsHandler,
+  getPostByIdHandler,
+  updatePostHandler,
+  deletePostByIdHandler,
+} from '../../../controllers/posts-controller.js';
+import Post from '../../../models/post.js';
+
+jest.mock('../../../models/post.js', () => ({
+  __esModule: true,
+  default: jest.fn(),
+}));
+
+const res = {
+  json: jest.fn(),
+  status: jest.fn().mockReturnThis(),
+};
+
+describe('createPostHandler', () => {
+  it('should create a new post', async () => {
+    const postObject = {
+      title: 'Test Post',
+      authorName: 'Test Author',
+      imageLink: 'https://www.forTestingPurposeOnly.com',
```

Comment:
A thing that caught my eye here:

The test should actually fail with this `imageLink`. This URL won't be accepted on the frontend because there's a regex that checks if the image URL ends with .png/.jpg/.jpeg.
See this line for reference: https://github.com/krishnaacharyaa/wanderlust/blob/2208b82b94f0932ec5cb6bfde1f62487f7881b8b/frontend/src/pages/add-blog.tsx#L63

Now, I get that the test will pass nevertheless because there is no check for this in the API handlers or in the model logic. We should open an issue for this and resolve it.

Reply:
Feel free to mark it resolved after opening an issue @hemant14050 or @krishnaacharyaa.

Reply:
That's a very nice catch @chinmaykunkikar, according to me we can solve it here in this PR as this is still not deployed in production, had it been already deployed we could have given fix for this by raising ticket, as it is other-way around we can fix it in this very PR, @hemant14050 kindly note the comments and make required changes,

For us the target is 3 checks of https://github.com/krishnaacharyaa/wanderlust/pull/48#pullrequestreview-1716115968, post these checks we can merge, otherwise it defeats the very purpose of adding test cases :)